### PR TITLE
Export quarterly yearly rent statement

### PR DESCRIPTION
**Pull request by EdwardAnoghena**
## Link to JIRA ticket
https://app.clickup.com/t/1kck5x1
https://app.clickup.com/t/1kck5wt

### *What is the problem we're trying to solve*
Ability to export quarterly  and yearly rent statement

### *What changes have we introduced*
Added Export Usecases
Include any links to commits that introduce significant additions of code if they help explain the process of coming to the solution. e.g Addition of test database setup, addition of first test and production class, removal of buggy code, etc.

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [ ] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [ ] Checked all code for possible refactoring
- [ ] Code pipeline builds correctly

### *Follow up actions after merging PR*

Are there any next steps that need to addressed after merging this PR? Add them here.

### CODE REVIEW

File: FinancialTransactionsApi/V1/Helpers/FileGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                   {
+                       Date = item.TransactionDate.ToString("dd MMM yyyy"),
+                       TransactionDetail = item.TransactionSource,
+                       Debit = Money.PoundSterling(item.PaidAmount).ToString(),
+                       Credit = Money.PoundSterling(item.HousingBenefitAmount).ToString(),
+                       Balance = Money.PoundSterling(item.BalanceAmount).ToString()
+                   });
+            }
+            report.Data = data;
+            var globalSettings = new GlobalSettings
```
Comment (LBHSPreston) [inline FinancialTransactionsApi/V1/Helpers/FileGenerator.cs:42]: These appear to be configurable items.  Should they not be included as environment variables should they change in the future?

File: FinancialTransactionsApi/V1/Helpers/FileGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            {
+                ColorMode = ColorMode.Color,
+                Orientation = Orientation.Portrait,
+                PaperSize = PaperKind.A4,
+                Margins = new MarginSettings { Top = 10 },
+                DocumentTitle = $"{name} Statement Report"
+            };
+            var objectSettings = new ObjectSettings
```
Comment (LBHSPreston) [inline FinancialTransactionsApi/V1/Helpers/FileGenerator.cs:50]: These appear to be configurable items. Should they not be included as environment variables should they change in the future?

File: FinancialTransactionsApi/V1/Helpers/FileGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                RentAccountNumber = _.PaymentReference,
+                Type = _.TransactionSource,
+                Charge = _.ChargedAmount,
+                Paid = _.PaidAmount,
+                HBCont = _.HousingBenefitAmount,
+                Balence = _.BalanceAmount
+            });
+
+            byte[] result;
+            var cc = new CsvConfiguration(new System.Globalization.CultureInfo("en-US"));
```
Comment (LBHSPreston) [inline FinancialTransactionsApi/V1/Helpers/FileGenerator.cs:117]: Why are we using en-US?  Should this not be en-UK?

File: FinancialTransactionsApi/V1/Helpers/TemplateGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        //	return sb.ToString();
+        //}
+
+
+        public static string GetHTMLReportString(ExportResponse report)
+        {
+            var date = DateTime.Today.ToString("D");
+            var sb = new StringBuilder();
+            sb.Clear();
+            sb.Append(@"<html><body>
```
Comment (LBHSPreston) [inline FinancialTransactionsApi/V1/Helpers/TemplateGenerator.cs:82]: What happens if this copy changes would it need a code update to be re-deployed?  I see that it is being built dynamically, would it be possible to use some sort of configurable template and inject the dynamic content?

**Review by LBHSPreston — COMMENTED** [review]
Added some comments.  Also wanted to know how the runtime binaries will get resolved when deployed.  Looks like this package looks for the binaries in a specific place.

File: FinancialTransactionsApi/V1/Helpers/FileGenerator.cs (hunk 1 above)
Comment (EdwardAnoghena) [reply FinancialTransactionsApi/V1/Helpers/FileGenerator.cs:42]: No, they are class settings that comes with the library "WkHtmlToPdfDotNet"

File: FinancialTransactionsApi/V1/Helpers/FileGenerator.cs (hunk 2 above)
Comment (EdwardAnoghena) [reply FinancialTransactionsApi/V1/Helpers/FileGenerator.cs:50]: the configurable item are already set-up on aws

**Review by LBHSPreston — APPROVED** [review]
Approved with the intention that hard coded values will be revisited as a subsequent task before it goes to production.

**Pull request merged**